src/language/branches.md

```diff
@@ -0,0 +1,229 @@
+matches that pattern. You can use [any kind of pattern][] for a case.
+
+Non-empty `case` clauses implicitly jump to the end of the switch after completion.
+Other valid ways to end a non-empty `case` clause are a [`continue`][break],
+[`throw`][], or [`return`][] statement.
+
+Use a `default` or wildcard `_` clause to execute code when no `case` clause matches:
+
+```dart
+var command = 'OPEN';
+switch (command) {
+  case 'CLOSED':
+    executeClosed();
+  case 'PENDING':
+    executePending();
+  case 'APPROVED':
+    executeApproved();
+  case 'DENIED':
+    executeDenied();
+  case 'OPEN':
+    executeOpen();
+  default:
+    executeUnknown();
+}
+```
+
+Empty cases fall through to the next case. For non-sequential fall-through,
+you can use a [`continue` statement][break] and a label:
+
+```dart
+switch (command) {
+  case 'OPEN':
+    executeOpen();
+    continue newCase;     // Continues executing at the newCase label.
+  
+  case 'DENIED':          // Empty case falls through.
+  case 'CLOSED':
+    executeClosed();      // Runs for both DENIED and CLOSED,
+  
+  newCase:
+  case 'PENDING':
+    executeNowClosed();   // Runs for both OPEN and PENDING.
+}
+```
+For an empty case that does not fall through, use [`break`][break] for its body.
+
+### Switch expressions
+
+Switch expressions are similar to switch statements, but you can use them anywhere
+you can use an expression. They produce a value based on the expression body of
+whichever case matches.
+
+Switch expression syntax differs slightly from switch statements:
+
+- Cases _do not_ start with the `case` keyword.
+- A case body is a single expression instead of a series of statements.
+- Each case must have a body; there is no implicit fallthrough for empty cases.
+- Case patterns are separated from their bodies using `=>` instead of `:`.
+- Cases are separated by `,` (and an optional trailing `,` is allowed).
+- Default cases can _only_ use `_`, instead of allowing both `default` and `_`.
+
+Switch expressions allow you to rewrite a _statement_ like this:
+
+```dart
+// Where slash, star, comma, semicolon, etc., are constant variables...
+
+switch (charCode) {
+  case slash || star || plus || minus:    // Logical-or pattern
+    token = operator(charCode);
+  case comma || semicolon:                // Logical-or pattern
+    token = punctuation(charCode);
+  case >= digit0 && <= digit9:            // Relational and logical-and patterns
+    token = number();
+  default:
+    throw invalid();
+}
+```
+
+Into an _expression_, like this:
+
+```dart
+token = switch (charCode) {
+  slash || star || plus || minus => operator(charCode),
+  comma || semicolon => punctuation(charCode),
+  >= digit0 && <= digit9 => number(),
+  _ => throw invalid()
+};
+```
+
+### Exhaustiveness checking
+
+Exhaustiveness checking is a feature of switches that reports a compile-time
+error if it's possible for a value entering the switch to not match any of the cases.
```

Comment:
this is one of those things I'm hesitant to change now that we're past technical reviews. Exhaustiveness checking happens _to_ switches... I know that saying it's `switch` that's responsible for exhaustiveness checking would not go over well; it would probably be the compiler or something, but it's weird to bring up the compiler in the language tour... I changed it, when you see the next commit see what you think